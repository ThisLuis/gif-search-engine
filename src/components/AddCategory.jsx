import { useState } from 'react';

export const AddCategory = ({ setCategories} ) => {

	const [ inputValue, setInputValue ] = useState('');

	const onInputChange = ( event ) => {
		setInputValue(event.target.value)
	}

	const onSubmit = ( event ) => {
		event.preventDefault();
		// inputValue es el valor que tenemos que mandar para agregar la categoria
		console.log( inputValue );
		if( inputValue.trim().length <= 1) return;
		setCategories( (categories) => [inputValue, ...categories]);
		setInputValue('');
	}

  return(
    <>
			<form onSubmit={ (event) => onSubmit(event) }>
				<input
					type="text"
					placeholder="Buscar gifs"
					value={ inputValue }
					onChange={ (event) => onInputChange(event) }
				/>
			</form>
    </>
  )
}
