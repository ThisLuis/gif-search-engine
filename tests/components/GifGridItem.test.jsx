import { render } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('Evaluando GifGridItem', () => {
	
	const title = 'One punch man';
	const url = 'www.google.com';

	test('evaluacion del snapshot', () => {
		const { container } = render(<GifGridItem title={title} url={ url } />)
		expect( container ).toMatchSnapshot();
	});

})