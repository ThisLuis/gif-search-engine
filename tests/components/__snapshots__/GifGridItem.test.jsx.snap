// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando GifGridItem evaluacion del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch man"
      src="www.google.com"
    />
    <p>
      One punch man
    </p>
  </div>
</div>
`;
